### PR TITLE
Refs #5691 -- Made cache keys independent of USE_L10N.

### PR DESCRIPTION
**Pull request by claudep**
This mostly reverts af1893c4ff8fdbf227a43a559d90bb1c1238b01a.

### CODE REVIEW

**Comment by claudep** [comment]
I'm working on reducing usage of USE_L10N (we may eventually get rid of it).

@aaugustin, you committed af1893c, but I didn't find any rationale in the related ticket. I know it's a long time, but could you explain what use case you were targeting?

The alternative would be to always include the language in cache keys, is that something that makes sense?

**Comment by aaugustin** [comment]
My best guess is that it's a matter of consistency between the three USE_* settings (I18N, L10N, TZ). When any of them is set, the cache key should be conditional on the locale or timezone.

As you're saying in the commit message, this only has an effect when I18N is False and L10N is True. Indeed, that's the scenario I added a test for in the commit we're talking about.

Since the default project template enables USE_I18N, it should be unfrequent that someone would disable it and then enable USE_L10N. But it isn't completely implausible.

A use case would be someone with an app in English that they don't have the resources to translate to various other European languages, but they'd still want to use localization (e.g. display dates as mm/dd in the US and dd/mm in the EU). If they wanted to avoid a mix of translated (by Django) and untranslated text (model names) in the admin, they could turn I18N off while keeping L10N.

Always including the language in cache keys could multiply the size of the cache by a large factor — the number of locales used by various users of the website — for websites that don't have I18N enabled. I'd rather avoid that option.

If possible, I would be more comfortable with one of the following options:

- going through a deprecation path (which will only be visible by the people who use caching, L10N but not I18N)
- keeping this until you finally axe USE_L10N, likely by merging it into USE_I18N

**Comment by claudep** [comment]
Thanks a lot for that explanation.

However I have some doubts about the `USE_I18N=False/USE_L10N=True` case. In that scenario, the `get_language()` call will always return the same `LANGUAGE_CODE`. So I don't see how this would work in practice. I suspect that `USE_L10N=True` requires `USE_I18N=True` to function properly.

**Comment by aaugustin** [comment]
It is quite likely that I missed this constraint!

**Comment by claudep** [comment]
In that case, I removed the release note from the patch.

**Comment by claudep** [comment]
@felixxm, may I ask you to put this in your review list at some point? Or do you think it's worth a ticket?

File: docs/topics/cache.txt [hunk 1]
```diff
@@ -542,9 +542,8 @@ include the name of the active :term:`language<language code>` -- see also
 :ref:`how-django-discovers-language-preference`). This allows you to easily
 cache multilingual sites without having to create the cache key yourself.
 
-Cache keys also include the active :term:`language <language code>` when
-:setting:`USE_L10N` is set to ``True`` and the :ref:`current time zone
-<default-current-time-zone>` when :setting:`USE_TZ` is set to ``True``.
+Cache keys also include the :ref:`current time zone <default-current-time-zone>`
+when :setting:`USE_TZ` is set to ``True``.
```
Comment (felixxm) [inline docs/topics/cache.txt:546]: As far as I'm concerned, previously when `USE_L10N` was `True` and `USE_I18N` was `False` then cache keys included `settings.LANGUAGE_CODE`. I think we should add release notes that in such configuration cache keys are different from the keys generated by older versions of Django :thinking:

Comment (claudep) [reply docs/topics/cache.txt:546]: By reading the code, it appears to me that the cache key is internal implementation. Do you see a way that this change could cause a backwards incompatibility?

Comment (felixxm) [reply docs/topics/cache.txt:546]: > By reading the code, it appears to me that the cache key is internal implementation. Do you see a way that this change could cause a backwards incompatibility?

No, but the first request to any previously cached value will be a cache miss. Am I right? :thinking: so maybe it's worth to add a short note (like we did in e.g. 5cb3ed187b2).

Comment (claudep) [reply docs/topics/cache.txt:546]: Fair, note added.

**Comment by felixxm** [comment]
@claudep Thanks :+1:

**Pull request merged**